### PR TITLE
Search view

### PR DESCRIPTION
**Pull request by noresjo**
Something quite useful as a start

### CODE REVIEW

File: client/landing_page.js [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
   self.reposRootPath = new ReactiveVar("");
+  self.searchText = new ReactiveVar("");
+  self.repoText = new ReactiveVar(Session.get("lastRepoName"));
+  self.actualRepo = new ReactiveVar("");
+
+  self.hideOverlay = function() {
+    self.$("#searchInput")
+      .val("")
+      .blur();
+    self.$(".overlay").hide();
```
Comment (lbergnehr) [inline client/landing_page.js]: Every `$(...)` will execute a new find. Don't know if jQuery caches or does anything else fancy, but shouldn't it be possible to chain those calls on the same element? E.g.:

    $(".landingPage").show().find("a").removeAttr("tabindex", "-1");

File: client/landing_page.js [hunk 2]
```diff
@@ -1,7 +1,48 @@
+var escapeRegExp = function(str) {
+  return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
+}
+
 Template.landingPage.created = function() {
   var self = this;
 
   self.reposRootPath = new ReactiveVar("");
+  self.searchText = new ReactiveVar("");
```
Comment (lbergnehr) [inline client/landing_page.js]: `searchText` doesn't have to be reactive or even an instance var at all does it?

Comment (noresjo) [inline client/landing_page.js]: So true. Will remove it.

File: client/landing_page.js (hunk 1 above)
Comment (noresjo) [inline client/landing_page.js]: Good idea. Fixing it.

**Comment by noresjo** [comment]
Anything else? We might want to tweak the UX-feel right?

**Comment by lbergnehr** [comment]
Yeah, but maybe that's for later?

**Comment by noresjo** [comment]
yeah... good enough maybe?

**Pull request merged**